### PR TITLE
[EAGLE-3010][EAGLE-3009]

### PR DESCRIPTION
**Pull request by steph-en-m**
* Add model upload support for text & visual classification + docs & examples
* Make improvements to model upload utils.

### CODE REVIEW

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/twitter-xlm-roberta-base-sentiment/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+language: multilingual
```
Comment (ackizilkale) [inline clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/twitter-xlm-roberta-base-sentiment/README.md:2]: nice, this is great that we provide a multilingual example here.

File: clarifai/models/model_serving/examples/image_classification/age_vit/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#clarifai>9.5.3 # for model upload features
```
Comment (ackizilkale) [inline clarifai/models/model_serving/examples/image_classification/age_vit/requirements.txt:1]: are we going to keep this one here?

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#clarifai>9.5.3 # for model upload features
```
Comment (ackizilkale) [inline clarifai/models/model_serving/examples/text_classification/xlm-roberta/requirements.txt:1]: and this one

File: clarifai/models/model_serving/examples/image_classification/age_vit/requirements.txt (hunk 2 above)
Comment (steph-en-m) [reply clarifai/models/model_serving/examples/image_classification/age_vit/requirements.txt:1]: yes, forgot to uncomment after tests.

File: clarifai/models/model_serving/model_config/triton_config.py [hunk 4]
```diff
@@ -147,5 +147,19 @@ def __post_init__(self):
       del pred_labels.labels
       pred_scores = OutputConfig(name="predicted_scores", data_type="TYPE_FP32", dims=[-1, 1])
       self.output.extend([pred_bboxes, pred_labels, pred_scores])
+
     elif self.model_type == "visual-classifier":
       self.input.append(image_input)
+      pred_labels = OutputConfig(
+          name="softmax_predictions", data_type="TYPE_FP32", dims=[-1], labels=True)
+      del pred_labels.labels
```
Comment (ackizilkale) [inline clarifai/models/model_serving/model_config/triton_config.py:155]: These labels deletion feels weird. When do we need labels field in OutputConfig?

Comment (steph-en-m) [reply clarifai/models/model_serving/model_config/triton_config.py:155]: some model types require labels.txt files while some don't.

File: clarifai/models/model_serving/docs/output.md [hunk 5]
```diff
@@ -16,3 +16,6 @@ from clarifai.models.model_serving.models.output import VisualDetectorOutput
 | | `predicted_labels` | int32 | [-1, 1] | A 2D detected labels array of length equal to that of predicted_bboxes with each element array having a length of exactly 1.
 | | `predicted_scores` | float32 | [-1, 1] | A 2D detection scores array of length equal to that of predicted_bboxes & predicted_labels with each element array having a length of exactly 1.
 | | | | | |
+| [VisualClassifierOutput](../models/output.py) | `predicted_scores` | float32 | [-1] | The softmax of the model's predictions. The index of each predicted probability as returned by the model must correspond to the label index in the labels.txt file |
+| | | | | |
+| [TextClassifierOutput](../models/output.py) | `predicted_scores` | float32 | [-1] | The softmax of the model's predictions. The index of each predicted probability as returned by the model must correspond to the label index in the labels.txt file |
```
Comment (HarmitMinhas96) [inline clarifai/models/model_serving/docs/output.md:21]: Can we just have ClassifierOutput? Afaik they are the same handling

File: clarifai/models/model_serving/examples/image_classification/age_vit/1/inference.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  -----
+    input_data: A single input data item to predict on.
+      Input data can be an image or text, etc depending on the model type.
+    model: Inference model callable as returned by setup() above.
+
+  Returns:
+  --------
+    One of the clarifai.model_serving.models.output types. Refer to the README/docs
+  """
+  transforms = ViTFeatureExtractor.from_pretrained(HUGGINGFACE_MODEL_PATH)
```
Comment (HarmitMinhas96) [inline clarifai/models/model_serving/examples/image_classification/age_vit/1/inference.py:52]: Seems slow to init this on every predict call. This seems to be similar issue to what we mentioned in the last PR with not having a class to handle this, where it can set 'self' variables.

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    input_data: A single input data item to predict on.
+      Input data can be an image or text, etc depending on the model type.
+    model: Inference model callable as returned by setup() above.
+
+  Returns:
+  --------
+    One of the clarifai.model_serving.models.output types. Refer to the README/docs
+  """
+  tokenizer_path = os.path.join(BASE_PATH, "twitter-xlm-roberta-base-sentiment")
+  tokenizer = AutoTokenizer.from_pretrained(tokenizer_path)
```
Comment (HarmitMinhas96) [inline clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py:54]: Similarly here

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  Load inference model.
+  The model checkpoint(s) should be saved in the same directory and directory
+  level as this inference.py module.
+
+  Returns:
+  --------
+    Inference Model Callable
+  """
+  checkpoint_path = os.path.join(BASE_PATH, "twitter-xlm-roberta-base-sentiment")
+  model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path)
```
Comment (HarmitMinhas96) [inline clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py:33]: Can we use the transformers pipelines which contain both model and tokenizer and are callable?

**Review by HarmitMinhas96 — COMMENTED** [review]
Can we consolidate classifiers since prediction handling is just repeated?

File: clarifai/models/model_serving/docs/model_types.md [hunk 9]
```diff
@@ -10,5 +10,5 @@ Clarifai [model types](../models/model_types.py) are decorator functions that ar
 ## Supported Model Types:
 
 - visual_detector
-- visual-classifier
-- text-classifier
+- visual_classifier
+- text_classifier
```
[the remark this answers is not in the document]
Comment (steph-en-m) [reply clarifai/models/model_serving/docs/model_types.md:14]: These are specific to the decorator types since the python coding style emphasizes snake case variable naming

File: clarifai/models/model_serving/docs/output.md (hunk 5 above)
Comment (steph-en-m) [reply clarifai/models/model_serving/docs/output.md:21]: sure

File: clarifai/models/model_serving/cli/repository.py [hunk 10]
```diff
@@ -50,10 +50,20 @@ def model_upload_init():
       default=".",  #curdir
       required=True,
       help="Directory to create triton repository.")
+  parser.add_argument(
+      "--use_gpu",
+      type=bool,
+      default=True,
+      required=False,
+      help="Specify whether to use GPU (True) or CPU (False). \
+        Defaults to GPU (True)")
```
[the remark this answers is not in the document]
Comment (steph-en-m) [reply clarifai/models/model_serving/cli/repository.py:59]: Had just changed that on my end. I had forgotten to make the change

**Review by ackizilkale — APPROVED** [review]
LGTM as long as the unresolved items are addressed.

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py (hunk 8 above)
Comment (steph-en-m) [reply clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py]: Pipeline returns class label strings so not using it to avoids extra ops of remapping to int and assigning to a score to a corresponding label index. 
Here I can just call softmax on the model output logits. (nb; temporarily defined other reusable inference time artifacts e.g. tokenizers, etc as global vars as I move to the class implementation)

File: clarifai/models/model_serving/examples/image_classification/age_vit/1/inference.py (hunk 6 above)
Comment (steph-en-m) [reply clarifai/models/model_serving/examples/image_classification/age_vit/1/inference.py]: eagle-3231

File: clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py (hunk 7 above)
Comment (steph-en-m) [reply clarifai/models/model_serving/examples/text_classification/xlm-roberta/1/inference.py]: eagle-3231

**Review by HarmitMinhas96 — APPROVED** [review]
